From 260bf50b2dfa826a7560bec2a5d9ba5b8b5ac25b Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumitrescu95@gmail.com>
Date: Mon, 10 Feb 2025 21:28:06 +0200
Subject: [PATCH 2/2] libtool.m4: Update NetBSD postdeps for shared libraries

* m4/libtool.m4: Linking a shared version of 'gcc' is not possible on
  NetBSD, since the shared version of the library is 'gcc_s'. postdeps
  have been updated to remove 'gcc' instances with sed.
---
 m4/libtool.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/libtool.m4 b/m4/libtool.m4
index c677ccfd..b57b8c31 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -7315,7 +7315,7 @@ if test yes != "$_lt_caught_CXX_error"; then
 	  _LT_TAGVAR(hardcode_shlibpath_var, $1)=no
 	fi
 	# Workaround some broken pre-1.5 toolchains
-	output_verbose_link_cmd='$CC -shared $CFLAGS -v conftest.$objext 2>&1 | $GREP conftest.$objext | $SED -e "s:-lgcc -lc -lgcc::"'
+	output_verbose_link_cmd='$CC -shared $CFLAGS -v conftest.$objext 2>&1 | $GREP conftest.$objext | $SED -e "s:--as-needed -lgcc_s --no-as-needed -lgcc -lc --as-needed -lgcc_s --no-as-needed -lgcc:--as-needed -lgcc_s:"'
 	;;
 
       *nto* | *qnx*)
-- 
2.45.2

